crates/cxx-qt-lib-extras/src/gui/qapplication.rs

```diff
@@ -0,0 +1,200 @@
+// SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Laurent Montel <laurent.montel@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::{QByteArray, QFont, QString, QStringList, QVector};
+use core::pin::Pin;
+
+#[cxx::bridge]
+mod ffi {
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/qbytearray.h");
+        type QByteArray = crate::QByteArray;
+        include!("cxx-qt-lib/qstring.h");
+        type QString = crate::QString;
+        include!("cxx-qt-lib/qstringlist.h");
+        type QStringList = crate::QStringList;
+        include!("cxx-qt-lib/qvector.h");
+        type QVector_QByteArray = crate::QVector<QByteArray>;
+        include!("cxx-qt-lib/qfont.h");
+        type QFont = crate::QFont;
+
+        include!("cxx-qt-lib-extras/qapplication.h");
+        type QApplication;
```

Comment:
So in #622 i needed to create a `QApplication` to do a widgets example. I also made a `QPushButton`.

Wonder if we add `QPushButton` to `cxx-qt-lib-extras` and then have an `examples/` folder in `cxx-qt-lib-extras` with the widgets example ? This could also then be a simple cargo-only thing to test building and CI ?

Reply:
For me it will be a good idea to add it + test